src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/WindowsForms/MyAppFileAccessor.cs

```diff
@@ -47,20 +65,34 @@ private void DocData_Modifying(object sender, EventArgs e)
 
     private async Task<MyAppDocument?> TryGetMyAppFileAsync()
     {
-        if (_docData is null)
+        if (_docData?.Data is null || !File.Exists(_absolutePath))
         {
-            string absolutePath = _project.MakeRooted("My Project\\Application.myapp");
-
             await _threadingService.SwitchToUIThread();
             try
             {
-                _docData = new DocData(_serviceProvider, absolutePath);
+                _docData = new DocData(_serviceProvider, _absolutePath);
                 _docData.Modifying += DocData_Modifying;
             } 
             catch (Exception)
             {
-                // If we have reached here, it means that the file doesn't exist in that location.
-                return null;
+                // If we have reached here, it means that the file doesn't exist in that location, so try to create it
```

Comment:
Generally, doing actual logic in a `catch` block is not a good design, especially writing files and other I/O. I'd do some adjustment in how we determine the myapp file needs created. This `catch` should only be for exceptional situations.